_ingest-pipelines/processors/dissect.md

```diff
@@ -0,0 +1,526 @@
+
+The dissection algorithm is precise and requires that every character in the pattern exactly match the source string. For instance, the pattern `%{hellokey} %{worldkey}` (one space) will match the string "Hello world" (one space) but not the string "Hello  world" (two spaces) because the pattern only has one space while the source string has two.
+
+The right padding modifier can be used to address this issue. By adding the right padding modifier to the pattern `%{helloworldkey->} %{worldkey}`, it will match <code>Hello&nbsp;world</code> (one space), <code>Hello&nbsp;&nbsp;world</code> (two spaces), and even <code>Hello&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;world</code> (ten spaces). 
+
+The right padding modifier is used to allow for the repetition of characters following a `%{keyname->}`. The right padding modifier can be applied to any key along with any other modifiers. It should always be the rightmost modifier, for example, `%{+keyname/1->}` or `%{}`.
+
+#### Example of usage
+
+The following is an example of how to use a right padding modifier:
+
+`%{city->}, %{state} %{zip}`
+
+In this pattern, the right padding modifier `->` is applied to the `%{city}` key. Both addresses contain the same information, but the second entry has an extra word, `City`, in the city field. The right padding modifier allows the pattern to match both of these address entries, even though they have slightly different formats: 
+
+```bash
+New York, NY 10017
+New York City, NY 10017
+```
+
+The following example pipeline uses the right padding modifier with an empty key `%{->}`:
+
+```json
+PUT /_ingest/pipeline/dissect-test
+{
+  "description": "Pipeline that dissects web server logs",
+  "processors": [
+    {
+      "dissect": {
+        "field": "message",
+        "pattern": "[%{client_ip}]%{->}[%{timestamp}]" 
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+You can test the pipeline using the following example:
+
+```json
+POST _ingest/pipeline/dissect-test/_simulate
+{
+  "docs": [
+    {
+      "_index": "testindex1",
+      "_id": "1",
+      "_source": {
+        "message": "[192.168.1.10]   [03/Nov/2023:15:20:45 +0000]"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+Your response should be similar to the following:
+
+```json
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "testindex1",
+        "_id": "1",
+        "_source": {
+          "client_ip": "192.168.1.10",
+          "message": "[192.168.1.10]   [03/Nov/2023:15:20:45 +0000]",
+          "timestamp": "03/Nov/2023:15:20:45 +0000"
+        },
+        "_ingest": {
+          "timestamp": "2024-01-22T22:55:42.090569297Z"
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Append modifier (`+`)
+
+The append modifier combines the values of two or more values into a single output value. The values are appended from left to right. You can also specify an optional separator to be inserted between the values. 
```

Comment:
Global: Should the names of the modifiers be in code font or otherwise formatted to indicate that they are names?

Reply:
In that case, it should be "An append modifier".